src/cloud_provider/aws/mod.rs

```diff
@@ -121,6 +125,10 @@ impl CloudProvider for AWS {
         }
     }
 
+    fn zones(&self) -> Vec<String> {
```

Comment:
Return a reference, you are always cloning and after modifying the zone, which re-duplicate memory